### PR TITLE
fix(Android): Avoid NullpointerException on install referer listener

### PR DESCRIPTION
**Pull request by nhnam**
action is Nullable, so we call equals from action is not safe

```
java.lang.RuntimeException · Unable to start receiver com.learnium.RNDeviceInfo.RNDeviceReceiver: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
```

## Description

Fixed issue #<issue-number>

Added `yourNewMethodName()` that allows ...

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅❌     |
| Android |    ✅❌     |
| Windows |    ✅❌     |

## Checklist

* [x] I have tested this on a device/simulator for each compatible OS
* [x] I added the documentation in `README.md`
* [x] I mentioned this change in `CHANGELOG.md`
* [x] I updated the typings files (`deviceinfo.d.ts`, `deviceinfo.js.flow`)
* [x] I updated the dummy web/test polyfill (`default/index.js`)
* [x] I added a sample use of the API (`example/App.js`)

### CODE REVIEW

**Comment by mikehardy** [comment]
I get your point about nullable action, and the proposed change is in general just good defensive coding practice, so I'm probably going to just merge this but I must ask :-), this actually happened to you? Because it should never happen, we filter for just that action so it should only ever be that action? https://github.com/react-native-community/react-native-device-info/blob/master/android/src/main/AndroidManifest.xml#L10

In some research it appears that someone is directly calling your action, which explains why it might start with the action being null: https://stackoverflow.com/questions/15048883/intent-getaction-is-returning-null

So I can believe this happens, and I think the patch is correct, I'm just surprised it's needed but hasn't come up before - it looks like it's more than a year old https://github.com/react-native-community/react-native-device-info/pull/344

**Comment by nhnam** [comment]
Hi, sorry for my bad fix. As you knew, the `action` is marked `@Nullable` so basically it could be null. We could either check for nullability first of use my way to just avoid the exception. 
Here goes I explain how could it happen. 
We're using `react-native-device-info` in our project and, another one which also has a receiver which is `react-native-appsflyer`
They notice here [https://support.appsflyer.com/hc/en-us/articles/207032126-AppsFlyer-SDK-Integration-Android](https://support.appsflyer.com/hc/en-us/articles/207032126-AppsFlyer-SDK-Integration-Android)
Please refer to #MultipleInstallBroadcastReceiver and #SingleInstallBroadcastReceiver. which I think it related to my issue, the NullPointerException.
So there are 2 options: 
`If you do not have a receiver listening on the INSTALL_REFERRER, use SingleInstallBroadcastReceiver` ==> we use this since we didn't know that `react-native-device-info` also has another Receiver. If we know that there are more BroadcastReceiver in the app, we would use the MultipleInstallBroadcastReceiver.

So, basically the appsflyer will catch INSTALL_REFERRER first then broadcast back to others Receiver (react-native-device-info), and in this case, the intent.getAction() is null.

**Comment by mikehardy** [comment]
makes sense! I'm happy to merge it, just wanted to understand more. I really appreciate the details on the explanation

**Comment by mikehardy** [comment]
You can point to master in your package.json until this is released - hopefully I can put this out shortly though

**Comment by nhnam** [comment]
Sure, thanks @mikehardy!

**Comment by mikehardy** [comment]
v2.1.1 just went out, thanks again for the contribution :smile:  :rocket:
